File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<Character>(); 

        for (char c : s.toCharArray()) { 
            if (c == '('){
                stack.push('('); 
            }
            else if (c == '{') 
            {
                stack.push('{'); 
            }
            else if (c == '['){
                stack.push('[');
            } 
            else if (!stack.isEmpty()){
                char bracket = stack.pop();
                switch(c){
                    case ']':
                       if(bracket != '[') return false;
                       break;
                    case '}':
                       if(bracket != '{') return false;
                       break;
                    case ')':
                       if(bracket != '(') return false;
                       break;
                }
                
            }
            else return false;
        }

        return stack.isEmpty();
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.kt
class Solution {
    fun isValid(s: String): Boolean {
        val stackS = mutableListOf<Char>()

        for(char in s){

            when(char){
                '(' ->  stackS.add(')')
                '[' ->  stackS.add(']')
                '{' -> stackS.add('}')
                else ->  
                    if(stackS.isNotEmpty()){
                        val poppedElement = stackS.removeLast()
                        if(poppedElement != char){
                            return false
                        }
                    }else return false 
            }

        }
        return stackS.isEmpty()
    }
}